### PR TITLE
virsh.pool_create: Rewirte and extend cases

### PR DESCRIPTION
**Pull request by will-Do**
BTW, this PR dependent on https://github.com/autotest/virt-test/pull/1939

Signed-off-by: Yanbing Du <ydu@redhat.com>

### CODE REVIEW

**Comment by will-Do** [comment]
```
#  ./run -t libvirt --test ydu-libvirt.virsh.pool_create -k
Running setup. Please wait...
SETUP: PASS (3.37 s)
DATA DIR: /home/ydu/Work/virt-test/shared/data
DEBUG LOG: /home/ydu/Work/virt-test/logs/run-2015-01-12-20.15.21/debug.log
TESTS: 11
(1/11) type_specific.ydu-libvirt.virsh.pool_create.positive_test.from_given_file: SKIP
(2/11) type_specific.ydu-libvirt.virsh.pool_create.positive_test.from_pre_def_pool.dir_pool: PASS (4.94 s)
(3/11) type_specific.ydu-libvirt.virsh.pool_create.positive_test.from_pre_def_pool.iscsi_pool: PASS (21.46 s)
(4/11) type_specific.ydu-libvirt.virsh.pool_create.positive_test.from_pre_def_pool.gluster_pool: PASS (9.34 s)
(5/11) type_specific.ydu-libvirt.virsh.pool_create.negative_test.invalid_extra_option: PASS (4.51 s)
(6/11) type_specific.ydu-libvirt.virsh.pool_create.negative_test.no_xml_file: PASS (4.39 s)
(7/11) type_specific.ydu-libvirt.virsh.pool_create.negative_test.invalid_xml_file: PASS (4.43 s)
(8/11) type_specific.ydu-libvirt.virsh.pool_create.negative_test.duplicate_pool_name: PASS (4.12 s)
(9/11) type_specific.ydu-libvirt.virsh.pool_create.negative_test.duplicate_pool_uuid: PASS (4.72 s)
(10/11) type_specific.ydu-libvirt.virsh.pool_create.negative_test.duplicate_pool_source: PASS (20.88 s)
(11/11) type_specific.ydu-libvirt.virsh.pool_create.negative_test.readonly_pool_create: PASS (4.49 s)
TOTAL TIME: 91.24 s (01:31)
TESTS PASSED: 10
TESTS FAILED: 0
TESTS SKIPPED: 1
SUCCESS RATE: 100.00 %

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool_create.py [hunk 1]
```diff
@@ -1,113 +1,76 @@
...
-    1) Use xml file from test parameters to create pool.
-    2) Dump the exist pool to create pool.
-    3) Negatively create pool(invalid option, no file, invalid file, duplicate
-       pool name, and create pool under readonly mode).
+    Create a libvirt pool from an XML file. The file could be given by tester or
+    generated by dumpxml a pre-defined pool.
     """
-
-    pool_xml = params.get("pool_create_xml_file")
+    pool_xml_f = params.get("pool_create_xml_file")
```
Comment (PandaWei) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool_create.py]: @will-Do
if user just remove  "/PATH/TO/POOL.XML" and forget to set valid xml file path.
The following check won't report any error.
if "/PATH/TO/POOL.XML" in pool_xml_f:
+ raise error.TestNAError("Please replace %s with valid pool xml file" %
+ pool_xml_f)

so,
In my opinion, it's better to set the default value of pool_xml_f like below,
pool_xml_f = params.get("pool_create_xml_file", "/PATH/TO/POOL.XML")

do you think so?

Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool_create.py]: Nice catch, I will update it, thanks

Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool_create.py]: Well, after look the code again, it's need to set the default value for ' pool_xml_f '. By default, if you don't set 'pool_xml_f', case will generate a file automatically, and that's what we want.

File: libvirt/tests/cfg/virsh_cmd/pool/virsh_pool_create.cfg [hunk 2]
```diff
@@ -5,7 +5,7 @@
     main_vm =
     start_vm = "no"
     pool_create_xml_file = "virt-test-pool.xml"
-    pool_create_name = "virt-test-pool"
+    pool_create_name = "virt-test-pool_1"
```
Comment (yumingfei) [inline libvirt/tests/cfg/virsh_cmd/pool/virsh_pool_create.cfg]: Why do you rename it with a suffix "_1"?

Comment (will-Do) [inline libvirt/tests/cfg/virsh_cmd/pool/virsh_pool_create.cfg]: Well, should be not reasonable. I will change it back, thanks

Comment (yumingfei) [inline libvirt/tests/cfg/virsh_cmd/pool/virsh_pool_create.cfg]: Well, I was just curious about it...

**Comment by yumingfei** [comment]
Thanks~

**Pull request merged**